src/services/search-builder.js

```diff
@@ -164,18 +164,15 @@ function SearchBuilder(model, opts, params, fieldNamesRequested) {
               const column = opts.sequelize.col(`${association.as}.${columnName}`);
 
               if (field.field === schemaAssociation.idField) {
+                let value = params.search;
+
+                // NOTICE: cast to Number to see if positive number
                 if (field.type === 'Number') {
-                  const value = parseInt(params.search, 10) || 0;
```

Comment:
```javascript
Number('18.9'); // returns 18.9
parseInt('18.9', 10); // returns 18
```
if such query does not fail in MSSQL, MySQL (it seems to work in PG):
```sql
select id
from users
where id = 18.9;
```
ok with the change.

Reply:
Thank you for this comment.
I tried with three databases sources: `postgres`, `mysql` and `mssql`; And I had no issue with floating number 👍 
Although a confirmation should be best.

Reply:
I will put back `parseInt` for this PR.
But I wonder... :thinking: We are in a specific case where `field.field === schemaAssociation.idField`, in human words, where we have a relation field.
It looks like it is possible to have primary key with float numbers although it is absolutely not recommended.
Using `parseInt` won't work with float primary key, `Number` will.
Regarding `Number('1e87')`, I don't think this use case will show up, as the user will look for specific number id, not its big representation.
But it should be in another Pull Request ;)